### ISSUE TITLE
Any plans for Qt 6?

### ISSUE THREAD

**Issue by VarLad** [JuliaGraphics/QML.jl#103]
Since Qt6 is the new major release of Qt, any plan to support that?

**Comment by barche** [comment]
Yes, already WIP on my local machine. It looks easy enough, luckily :)

**Comment by clad26** [comment]
What about the binaries/artifacts？

**Comment by zmactep** [comment]
Hi, @barche!

Thanks for the great project. So, do you still have a plan to release any Qt6 version of the library?

**Comment by barche** [comment]
There is now a `qt6` branch in this repository as well as in the jlqml and QmlJuliaExamples repositories. This experimental work relies on JLL packages that can be installed after adding https://github.com/barche/CxxWrapTestRegistry.git as a registry (Linux and Mac only).

Next steps:

- [ ] Publish the Qt6* JLL build scripts to Yggdrasil
- [ ] Implement a TableModel, since dynamic columns in a ListView are not supported anymore in Qt6

**Comment by barche** [comment]
With the first step done, it is now fairly easy to test Qt6 with the examples. First clone the example `qt6` branch and start Julia in its directory:

```
git clone --branch qt6 https://github.com/barche/QmlJuliaExamples
cd QmlJuliaExamples
julia --project
```

I Julia pkg mode, make sure to switch to the Qt 6 version of QML:

```
add QML#qt6
```

And then run e.g. the gui.jl example:

```julia
include("gui.jl")
```

**Comment by zmactep** [comment]
Great, works for me!

**Comment by barche** [comment]
The Qt 6 changes are now merged into main. To test these, you need to add my test registry to get the correct shared library JLL:

```
registry add https://github.com/barche/CxxWrapTestRegistry.git
```

**Comment by barche** [comment]
I had hoped the Qt6 version would be released by now, but I hit a problem running the tests with Qt 6.4, see description here: https://forum.qt.io/post/746459

**Comment by ufechner7** [comment]
Perhaps this can be closed now that 0.8 is released?
